bundles/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/se/handler/ValloxSEHandler.java

```diff
@@ -0,0 +1,479 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.vallox.internal.se.handler;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.vallox.internal.se.cache.ValloxExpiringCacheMap;
+import org.openhab.binding.vallox.internal.se.configuration.ValloxSEConfiguration;
+import org.openhab.binding.vallox.internal.se.connection.ValloxConnector;
+import org.openhab.binding.vallox.internal.se.connection.ValloxEventListener;
+import org.openhab.binding.vallox.internal.se.mapper.ChannelDescriptor;
+import org.openhab.binding.vallox.internal.se.telegram.Telegram;
+import org.openhab.binding.vallox.internal.se.telegram.TelegramFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ValloxSEHandler} is responsible for handling commands, which
+ * are sent to one of the channels
+ *
+ * @author Hauke Fuhrmann - Initial contribution
+ * @author Miika Jukka - Rewrite
+ */
+@NonNullByDefault
+public class ValloxSEHandler extends BaseThingHandler implements ValloxEventListener {
+
+    private final Logger logger = LoggerFactory.getLogger(ValloxSEHandler.class);
+
+    private final ValloxExpiringCacheMap cache = new ValloxExpiringCacheMap(Duration.ofMinutes(15));
+    private final ValloxConnector connector;
+    private @NonNullByDefault({}) ScheduledFuture<?> refreshJob;
+    private @NonNullByDefault({}) ScheduledFuture<?> watchDog;
```

Comment:
please make these nullable instead of using `@NonNullByDefault({})`

Reply:
Done.